src/atom_vec.cpp

```diff
@@ -214,30 +214,30 @@ void AtomVec::grow(int n)
     cols = mgrow.cols[i];
     if (datatype == DOUBLE) {
       if (cols == 0)
-        memory->grow(*((double **) pdata),nmax*threads[i],"atom:dvec");
+        memory->grow(*((double **) pdata),nmax*(threads[i]?nthreads:1),"atom:dvec");
```

Comment:
How about moving `nthreads` into the loop and setting it to 1 `if !threads[0] == false`. That way you avoid multiple `COND ? true-expr : false-expr` constructs

Reply:
i don't think it can be avoided, since any threads[i] entry may be true or false. see https://github.com/lammps/lammps/pull/2080/files#diff-5b00f9a402c86c8735a80d7067e68582R2390

Reply:
Sorry, had a typo. Of course you have to check threads[i] for each loop iteration. What I mean is that for each iteration you don't have to duplicate that check X times, just have the result set once.

Reply:
```
for (int i = 0; i < ngrow; i++) {
   const int nthreads = threads[i] ? nthreads : 1;
   ...
}
```

Reply:
ok. it doesn't make a difference in terms of performance, since only one of the tests is run per iteration, but it improves readability.